R-packages/evalcast/R/signal_listcols.R

```diff
@@ -0,0 +1,39 @@
+# In the signals tibble, required by `get_predictions()`,
+# start_day` and `as_of` can possibly be functions
+# of the forecast_date. Here we evaluate those functions as well as creating
+# default values for `geo_values` and `geo_type` if necessary
+signal_listcols <- function(signals, forecast_date) {
+  
+  cn <- colnames(signals)
+  ns <- nrow(signals)
+  # start_day should be NULL or date. We handle date/chr/fun(forecast_date) input
+  if ("start_day" %in% cn && is.list(signals$start_day)) {
+    std <- Date(ns)
+    for (i in 1:ns) {
```

Comment:
I don't think it's an issue here, but consider using `seq_len` when creating sequences. It avoids issues with objects that can be length 0.
e.g. `for (i in 1:0)` loops through i = 1 and i =0, but `for i in seq_len(0)` doesn't.

Reply:
I always forget to use that. Done.